tests/test_authz_authorizer.py

```diff
@@ -133,81 +244,27 @@ def test_refs(self):
-        self.authz.deny = {"test_resource": {"permission_name": ALL}}
-        with pytest.raises(PermissionDenied):
-            self.authz._check_deny()  # pylint: disable=protected-access
-        assert self.authz.outcome == DENY
-
-    def test_check_deny_one_scope(self):
-        self.authz.deny = {"test_resource": {"permission_name": "self"}}
-        self.authz._check_deny()  # pylint: disable=protected-access
-        assert self.authz.denied_resource == "self"
-
-    def test_allow_if_all(self):
-        assert self.authz._allow_if_allow_all(ALL)  # pylint: disable=protected-access
-        assert self.authz.outcome == ALLOW
-        assert self.authz.allowed_resource == ALL
-
-    def test_check_allow_all(self):
-        self.authz._check_allow_and_set_resources()  # pylint: disable=protected-access
-        assert self.authz.outcome == ALLOW
-
-    def test_check_allow_fail_all(self):
-        self.authz.allow = None
-        with pytest.raises(PermissionDenied):
-            self.authz._check_allow_and_set_resources()  # pylint: disable=protected-access
-            assert self.authz.outcome == ALLOW
-
-    def test_check_allow_one(self):
-        self.authz.allow = {"test_resource": {"permission_name": {"allow": ALL}}}
-        self.authz._check_allow_and_set_resources()  # pylint: disable=protected-access
-        assert self.authz.outcome == ALLOW
-
-    def test_check_allow_one_scope(self):
-        self.authz.allow = {"test_resource": {"permission_name": {"allow": "self"}}}
-        self.authz._check_allow_and_set_resources()  # pylint: disable=protected-access
-        assert self.authz.outcome == ALLOW
-        assert self.authz.allowed_resource == "self"
-
-    def test_check_allow_fail_one_scope(self):
-        self.authz.allow = {"test_resource": {"permission_name": {"deny": "self"}}}
-        self.authz._check_allow_and_set_resources()  # pylint: disable=protected-access
-        assert self.authz.outcome == ALLOW
-        assert self.authz.denied_resource == "self"
-
-    def test_get_restrictions_restricted(self):
-        self.authz.allowed_resource = ALL
-        self.authz.denied_resource = {"city": "warszawa"}
-        assert self.authz.restrictions == {"allow": ALL, "deny": {"city": "warszawa"}}
-
-    def test_sign_authz(self):
+    def test_sign_authz(self) -> None:
         token = Authorizer.sign_authz({"allow": {ALL: ALL}, "deny": {}}, SAMPLE_PRIVATE_KEY)
         assert token == EXPECTED_TOKEN
+
+    def test_sign_authz_not_a_dict_error(self) -> None:
+        with pytest.raises(ValueError) as execinfo:
+            Authorizer.sign_authz({}, private_key_jwk="")
+            assert str(execinfo.value) == "private_key_jwk must be a jwk dict"
+
+    def test_sign_authz_no_kid_error(self) -> None:
+        with pytest.raises(ValueError) as execinfo:
+            Authorizer.sign_authz({}, private_key_jwk={})
+            assert str(execinfo.value) == "private_key_jwk must have the 'kid' field"
```

Comment:
Pytest provides the possibility to check the error message in a more elegant way:
```python
with pytest.raises(ValueError, match="private_key_jwk must have the 'kid' field"):
    Authorizer.sign_authz({}, private_key_jwk={})
```